books/RayTracingInOneWeekend.html

```diff
@@ -3477,50 +3477,47 @@
 
 Modeling a Hollow Glass Sphere
 -------------------------------
-An interesting and easy trick with dielectric spheres is to note that if you use a negative radius,
-the geometry is unaffected, but the surface normal points inward.
+Let's model a hollow glass sphere. This is a sphere of some thickness with another sphere of air
+inside it. If you think about the path of a ray going through such an object, it will hit the outer
+sphere, refract, hit the inner sphere (assuming we do hit it), refract a second time, and travel
+through the air inside. Then it will continue on, hit the inside surface of the inner sphere,
+refract back, then hit the inside surface of the outer sphere, and finally refract and exit back
+into the scene atmosphere.
 
-However, properly handling negative radii can be tricky. Recall the line from `sphere::hit()` in
-listing [sphere-material] that calculates the outward normal:
+The outer sphere is just modeled with a standard glass sphere, with an index of refraction of around
+1.50. The inner sphere is a bit different, because _its_ index of refraction will now be relative to
```

Comment:
Some optional small suggestions for this paragraph in **_bold_**:

The outer sphere is just modeled with a standard glass sphere, with an index of refraction of around 1.50 **_relative to the surrounding air_**. The inner sphere is a bit different because _its_ index of refraction **_should be_** relative to the **_surrounding glass outer sphere and model a transition back to air_**. This is actually simple ...

Reply:
I'll tweak the wording.

Reply:
Hmmm, I barely tweaked it:
> The outer sphere is just modeled with a standard glass sphere, with an index of refraction of around
1.50. The inner sphere is a bit different, because _its_ index of refraction should be relative to
the "glass atmosphere" of the **surrounding** outer sphere. This is actually simple to specify, as our

I don't want to add "relative to the surrounding air" since it's really a surrounding vacuum, (air = 1.000293), and most refractive indices are given with an implicit understanding that they're already relative to vacuum. Also, for the second change it's actually transitioning back into glass.

Either way, I'm going to add a bit more explanation earlier where we introduce refraction.

Reply:
I agree, vacuum is more correct. I only suggested air because it was used elsewhere in the book. My main point was to highlight that the refractive index is always relative to the surrounding medium (while a reader may assume its _absolute_ meaning, i.e. relative to vacuum). I think mentioning "relative to the surrounding air/vacuum" in the first sentence sets things up for the second sentence but please use whichever phrasing you prefer.

> ... Also, for the second change it's actually transitioning back into glass.

Doesn't a transition from the inner sphere's exterior (as defined by the outward facing normals) to its interior represent a transition from glass to air/vacuum (or whatever interior/exterior medium pair that happens to have the same refraction ratio)? That's really what I meant but I may have misunderstood something.

> Either way, I'm going to add a bit more explanation earlier where we introduce refraction.

Sounds good. As I said I think the main concern was to highlight that the refractive index is always relative. Even the absolute one is relative but readers may make assumptions about the definition of IOR and get confused about the meaning of `dialectric::ref_index`.

Reply:
See the next commit and let me know what you think.

> Doesn't a transition from the inner sphere's exterior (as defined by the outward facing normals) to its interior represent a transition from glass to air/vacuum (or whatever interior/exterior medium pair that happens to have the same refraction ratio)? That's really what I meant but I may have misunderstood something.

Ah, yes, I misunderstood your suggested reading. I think it was the "back to air" that confused me. I'd rather explain this as a transition into a third medium that just happens in this example to be the same as the first medium. Anyway, I think the paragraph immediately preceding this explains this all in detail, and we don't need to repeat it in this paragraph.

Reply:
This?

> The outer sphere is just modeled with a standard glass sphere, with an index of refraction of around 1.50 relative to the surrounding air, **modeling a refraction from air into glass**. The inner sphere is a bit different because its index of refraction should be relative to the surrounding glass outer sphere and model a transition **from glass into air**. This is actually simple ...

Reply:
Kind of wish we had three different materials to keep things straight, but the model itself would then be rather needlessly complicated. Perhaps use the terms "outside air" and "inside air"? `¯\_(ツ)_/¯`